opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap.py

```diff
@@ -48,18 +54,21 @@ def wrapper(package=None):
 @_syscall
 def _sys_pip_install(package):
     # explicit upgrade strategy to override potential pip config
-    subprocess.check_call(
-        [
-            sys.executable,
-            "-m",
-            "pip",
-            "install",
-            "-U",
-            "--upgrade-strategy",
-            "only-if-needed",
-            package,
-        ]
-    )
+    try:
+        check_call(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "install",
+                "-U",
+                "--upgrade-strategy",
+                "only-if-needed",
+                package,
+            ]
+        )
+    except CalledProcessError as error:
```

Comment:
To clarify the experience, `bootstrap_gen.py` will now contain the generated instrumentations from the local directories (which may include unreleased instrumentations). When users use `opentelemetry-bootstrap`, the logic will run this code, and the logic iterates through the generated file but instead of throwing an exception, we catch it and print the error instead correct?

Reply:
That handling is what I'm seeing with `opentelemetry-bootstrap` with this change (version changed to `0.45b0`). The logs include these so the messages are still visible:

```
ERROR: Could not find a version that satisfies the requirement opentelemetry-instrumentation-threading==0.45b0 (from versions: none)
ERROR: No matching distribution found for opentelemetry-instrumentation-threading==0.45b0
Command '['/usr/local/bin/python', '-m', 'pip', 'install', '-U', '--upgrade-strategy', 'only-if-needed', 'opentelemetry-instrumentation-threading==0.45b0']' returned non-zero exit status 1.

ERROR: Could not find a version that satisfies the requirement opentelemetry-instrumentation-aiohttp-server==0.45b0 (from versions: none)
ERROR: No matching distribution found for opentelemetry-instrumentation-aiohttp-server==0.45b0
Command '['/usr/local/bin/python', '-m', 'pip', 'install', '-U', '--upgrade-strategy', 'only-if-needed', 'opentelemetry-instrumentation-aiohttp-server==0.45b0']' returned non-zero exit status 1.
```

Bootstrapping does not crash; no more traceback with uncaught `subprocess.CalledProcessError`. 👍 👍  I can then `opentelemetry-instrument` my test service and the other, successfully-installed instrumentors generate spans.

Reply:
> To clarify the experience, `bootstrap_gen.py` will now contain the generated instrumentations from the local directories (which may include unreleased instrumentations). When users use `opentelemetry-bootstrap`, the logic will run this code, and the logic iterates through the generated file but instead of throwing an exception, we catch it and print the error instead correct?

That's right :v: